pkg/agent/plugin/workloadattestor/k8s/k8s_posix.go

```diff
@@ -199,8 +239,19 @@ func (p *Plugin) Attest(ctx context.Context, req *workloadattestorv1.AttestReque
 			status, lookup := lookUpContainerInPod(containerID, item.Status)
 			switch lookup {
 			case containerInPod:
+				selectors := getSelectorValuesFromPodInfo(&item, status)
+				if p.config.EnableSigstore {
+					log.Debug("Attemping to get signature info for container", telemetry.ContainerName, status.Name)
+					sigstoreSelectors, err := p.sigstore.AttestContainerSignatures(ctx, status)
+					if err != nil {
+						log.Error("Error retrieving signature payload: ", "error", err)
```

Comment:
Although we have a few exceptions, we have traditionally followed an "all or nothing" approach, where we return an error if there is a failure collecting one of the selectors. Otherwise, it's difficult to assert why the selector was not included. Was it omitted because there was an error or because it was not discovered? 
Particularly in this case, since the user explicitly enabled the sigstore functionality to get to this point, I would suggest that we fail if there is an error retrieving the signature payload.